vividus-bdd-engine/src/main/java/org/vividus/bdd/steps/VariableResolver.java

```diff
@@ -33,7 +36,16 @@
     public VariableResolver(IBddVariableContext bddVariableContext, Map<String, DynamicVariable> dynamicVariables)
     {
         this.bddVariableContext = bddVariableContext;
-        this.dynamicVariables = dynamicVariables;
+        this.dynamicVariables = new HashMap<>(dynamicVariables.size() * 2);
```

Comment:
let java do its work: if you have a single-word variable, then the size won't be x2

Reply:
but at the same time it won't be 16 buckets for 1 item